homeassistant/components/insteon/utils.py

```diff
@@ -383,18 +386,36 @@ def print_aldb_to_log(aldb):
 
 @callback
 def async_add_insteon_entities(
-    hass, platform, entity_type, async_add_entities, discovery_info
-):
-    """Add Insteon devices to a platform."""
-    new_entities = []
-    device_list = [discovery_info.get("address")] if discovery_info else devices
+    hass: HomeAssistant,
+    platform: Platform,
+    entity_type: type,
+    async_add_entities: Callable,
```

Comment:
```suggestion
    async_add_entities: AddEntitiesCallback,
```

Reply:
```suggestion
    entity_type: type[InsteonEntity],
    async_add_entities: AddEntitiesCallback,
```
I think `entity_type` could also be annotated more explicitly.